sdcm/nemesis.py

```diff
@@ -687,6 +798,141 @@ def _modify_table_property(self, name, val, filter_out_table_with_counter=False,
+            for result_rows in query_result:
+                keyspaces.extend([row.lower() for row in result_rows])
+        self.log.warn('_get_all_non_compact_tables keyspaces = [%s]' % (str(keyspaces),))
+        output = {}
+        for ks in keyspaces:
+            tables = get_tables_with_compaction(self.loaders.nodes[0], self.target_node, ks, non_compact=True)
+            if not tables:
+                continue
+            if allowed:
+                if allowed[ks]:
+                    tmp = []
+                    for table in tables:
+                        if table in allowed[ks]:
+                            tmp.append(table)
+                    if tmp:
+                        output[ks] = tmp
+            else:
+                output[ks] = tables
+        return output
+
+    def _get_target_noncompact_table(self, stored_target_table: list):
+        current_tables = self._get_all_non_compact_tables()
+        if stored_target_table:
+            if stored_target_table[1] in current_tables.get(stored_target_table[0], []):
+                return stored_target_table
+        if not current_tables:
+            return None
+        ks_name = current_tables.keys()[0]
+        table_name = current_tables[ks_name][0]
+        stored_target_table.clear()
+        stored_target_table.append(ks_name)
+        stored_target_table.append(table_name)
+        return stored_target_table
+
+    @staticmethod
+    def _get_added_columns_info(target_table: list, added_fields):
+        ks = added_fields.get(target_table[0], None)
+        if ks is None:
+            output = {'column_names': {}, 'column_types': {}}
+            ks[target_table[0]] = {target_table[1]: output}
+            return output
+        table = ks.get(target_table[1], None)
+        if table is not None:
+            return table
+        ks[target_table[1]] = output = {'column_names': {}, 'column_types': {}}
+        return output
+
+    @staticmethod
+    def _random_column_name(avoid_names=None, max_name_size=5):
+        if avoid_names is None:
+            avoid_names = []
+        while True:
+            column_name = ''.join(random.choice(string.ascii_uppercase + string.digits) for x in range(max_name_size))
+            if column_name not in avoid_names:
+                break
+        return column_name
+
+    def _add_remove_column(self, remove=True, add=True):
+        added_columns_info = self._get_added_columns_info(self._add_remove_column_target_table)
+        added_columns_info.get('column_names', None)
+        if not added_columns_info['column_names']:
+            remove = False
+        CMD = "ALTER TABLE " + self._add_remove_column_target_table[1]
+        if remove:
+            remove = []
+            for n in range(random.randrange(1,min(
+                    len(added_columns_info['column_names']), self._add_remove_column_max_per_remove
+            ))):
+                while True:
```

Comment:
Those `while True` loop tend to bite hard. should have the stopping condition in it, and not in inside of the loop.